src/main/java/net/sf/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -1296,4 +1297,31 @@ public int hashCode() {
 
     }
 
+
+    public BibDatabase generateDatabase(List<BibDatabase> databases, OOBibStyle style)
+            throws NoSuchElementException, WrappedTargetException {
+        BibDatabase resultDatabase = new BibDatabase();
+        List<String> cited = findCitedKeys();
+        for (String key : cited) {
+            for (BibDatabase loopDatabase : databases) {
+                Optional<BibEntry> entry = loopDatabase.getEntryByKey(key);
+                if (entry.isPresent()) {
+                    BibEntry clonedEntry = (BibEntry) entry.get().clone();
+                    clonedEntry.setId(IdGenerator.next());
+                    resultDatabase.insertEntry(clonedEntry);
+                    entry.get().getFieldOptional(FieldName.CROSSREF).ifPresent(crossref -> {
```

Comment:
The break solved the problem with the multiple entries. I was really confused for a while, but since the function looks through all tabs, it found the first generated database and included those entries once more and so on... :-)